### PR TITLE
Update cache when project is updated

### PR DESCRIPTION
**Pull request by FrankApiyo**
Fixes https://github.com/onaio/onadata/issues/1875
### Changes / Features implemented
- Update cache when a project is updated
### Steps taken to verify this change does what is intended
- Tested locally and added unit tests

### CODE REVIEW

File: onadata/apps/api/tests/viewsets/test_project_viewset.py [hunk 1]
```diff
@@ -1149,6 +1149,29 @@ def test_project_partial_updates(self):
...
+        data = {'public': True}
+        request = self.factory.patch('/', data=data, **self.extra)
+        response = view(request, pk=projectid)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(True, response.data.get("public"))
+
+        request = self.factory.get('/', **self.extra)
+        response = view(request, pk=self.project.pk)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(True, response.data.get("public"))
```
Comment (ivermac) [inline onadata/apps/api/tests/viewsets/test_project_viewset.py:1173]: I'd like to think we'd want to test caching by comparing response.data with `cache.get(f'{PROJ_OWNER_CACHE}{project_id}')` which is found in the retrieve action.

Comment (FrankApiyo) [reply onadata/apps/api/tests/viewsets/test_project_viewset.py:1177]: I added some assertions/comparisons for cached data and response data

**Comment by ivermac** [comment]
@FrankApiyo as usual, could you please create an issue for this pull request

**Comment by FrankApiyo** [comment]
> @FrankApiyo like always, could you please create an issue for this pull request

@ivermac done!

**Pull request merged**